R/utils.R

```diff
@@ -12,10 +12,17 @@ is_pipe <- function(x) {
 .infixes <- c("+", "-", "*", "/", "^", "%%", "%/%", "%in%", .infixes_assign)
 
 is_infix <- function(x, infix_vals = .infixes) {
-  if (is.character(x)) x <- parse(text = x)[[1]]
-  ifelse(is.call(x), as.character(x[[1]]) %in% infix_vals, FALSE)
+  
+  # if str2lang fails, x is not an infix
+  out <- x
+  if (is.character(x)){out<-"";  try(out <- str2lang(x), silent=TRUE) } 
+  # if (is.character(x)) x <- parse(text = x)[[1]]
+  ifelse(is.call(out), as.character(out[[1]]) %in% infix_vals, FALSE)
```

Comment:
Cosmetic changes (untested)

```suggestion
  tryCatch({
    if (is.character(x)) {
      # if str2lang throws, x is not an infix
      out <- str2lang(x)
    } else {
      out <- x
    }
    
    if (!is.call(out)) {
      return(FALSE)
    }
    
    as.character(out[[1]]) %in% infix_vals
  }, error = function(e) {
    # x is not an infix
    FALSE
  })
```